src/components/ha-tabs.js

```diff
@@ -0,0 +1,57 @@
+import "@polymer/paper-tabs";
+
+const PaperTabsClass = customElements.get("paper-tabs");
+let subTemplate;
+
+class HaTabs extends PaperTabsClass {
+  static get template() {
+    if (!subTemplate) {
+      subTemplate = PaperTabsClass.template.cloneNode(true);
+
+      const superStyle = subTemplate.content.querySelector("style");
+
+      superStyle.appendChild(
+        document.createTextNode(`
+          .hidden, .not-visible {
+            display: none;
+          }
+        `)
+      );
+    }
+    return subTemplate;
+  }
+
+  ready() {
+    super.ready();
+    this.setScrollDirection("y", this.$.tabsContainer);
+
+    this.addEventListener("mousewheel", (e) => {
+      if (e.wheelDelta > 0) {
+        this._scrollToRight();
+      } else if (e.wheelDelta < 0) {
+        this._scrollToLeft();
+      } else {
+        return;
+      }
+      e.preventDefault();
+    });
+  }
+
+  _affectScroll(dx) {
```

Comment:
Can you explain in a comment why we adjust this function?

Reply:
When I adjusted the scroll tabs to take up no more space than the actual tab container, it meant that the tab container actually shrinks when the scroll arrows appear. When that happens the area that you were scrolling jumps by the width of the scroll arrow. This prevents that from happening by jumping in the opposite direction the same amount, giving the appearance that nothing has changed and the scroll is smooth.